Framework/Core/include/Framework/ASoA.h

```diff
@@ -2521,7 +2536,33 @@ class Filtered : public FilteredBase<T>
   {
     return operator*=(other.getSelectedRows());
   }
-  using FilteredBase<T>::sliceByCached;
+
+  auto sliceByCached(framework::expressions::BindingNode const& node, int value)
+  {
+    uint64_t offset = 0;
+    std::shared_ptr<arrow::Table> result = nullptr;
+    if (!((table_t*)this)->getSliceFor(value, node.name.c_str(), result, offset).ok()) {
+      o2::framework::throw_error(o2::framework::runtime_error("Failed to slice table"));
+    }
+    if (offset >= this->tableSize()) {
+      self_t fresult{{result}, SelectionVector{}, 0}; // empty slice
+      this->copyIndexBindings(fresult);
+      return fresult;
+    }
+    auto start = offset;
+    auto end = start + result->num_rows();
+    auto start_iterator = std::lower_bound(this->getSelectedRows().begin(), this->getSelectedRows().end(), start);
+    auto stop_iterator = std::lower_bound(start_iterator, this->getSelectedRows().end(), end);
+    SelectionVector slicedSelection{start_iterator, stop_iterator};
+    std::transform(slicedSelection.begin(), slicedSelection.end(), slicedSelection.begin(),
+                   [&](int64_t idx) {
```

Comment:
In general try to avoid `[&]` but specify the actual elements you want to capture. I guess `[start]` here.